cloudify_cli/commands/cluster.py

```diff
@@ -89,15 +90,37 @@ def _deco(f):
         @cfy.pass_client(*client_args, **client_kwargs)
         @wraps(f)
         def _inner(client, *args, **kwargs):
-            managers_list = client.manager.get_managers().items
-            if len(managers_list) == 1:
-                get_logger().warning('It is highly recommended to have more '
-                                     'than one manager in a Cloudify cluster')
+            if _all_in_one_manager(client):
+                raise CloudifyCliError('You cannot run cluster actions on an '
+                                       'all-in-one Manager')
             return f(client=client, *args, **kwargs)
         return _inner
     return _deco
 
 
+def _all_in_one_manager(client):
+    manager_nodes = client.manager.get_managers().items
+    if len(manager_nodes) > 1:
+        return False
+    broker_nodes = client.manager.get_brokers().items
+    if len(broker_nodes) > 1:
+        return False
+    db_nodes = client.manager.get_db_nodes().items
+    if len(db_nodes) > 1:
+        return False
```

Comment:
Why are the added lines necessary for the assertion?

Reply:
I figured out short-circuiting the check with as few rest calls as possible would be a good thing